datasource/ethereum/src/block_stream.rs

```diff
@@ -56,6 +59,12 @@ enum BlockStreamState {
     /// Valid next states: Reconciliation
     YieldingBlocks(Box<Stream<Item = EthereumBlockWithTriggers, Error = Error> + Send>),
 
+    /// The BlockStream experienced an error and is pausing before attempting to produce
+    /// blocks again.
+    ///
+    /// Valid next states: Reconciliation
+    Paused(Box<Future<Item = (), Error = Error> + Send>),
```

Comment:
Paused sounds like it could also be paused for a long time, like being put on halt. Maybe `RetryAfterDelay`?